package/gluon-core/luasrc/lib/gluon/upgrade/820-dns-config

```diff
@@ -25,22 +25,23 @@ if dns.servers then
 	})
 end
 
-if next_node.name and next_node.ip4 then
-	uci:section('dhcp', 'domain', 'nextnode4', {
-		name = next_node.name,
-		ip = next_node.ip4,
+local function set_dns_record(name, ip, sectionname)
+	if not ip then return end
+	uci:section('dhcp', 'domain', sectionname, {
+		name = name,
+		ip = ip,
 	})
-else
-	uci:delete('dhcp', 'domain', 'nextnode4')
 end
 
-if next_node.name and next_node.ip6 then
-	uci:section('dhcp', 'domain', 'nextnode6', {
-		name = next_node.name,
-		ip = next_node.ip6,
-	})
-else
-	uci:delete('dhcp', 'domain', 'nextnode6')
+uci:foreach("dhcp", "domain", function(s)
+	if (s['.name'] and string.match(s['.name'], "^nextnode[46]_")) then
+		uci:delete("dhcp", s['.name'])
+	end
```

Comment:
Hmm, somehow one of my comments disappeared when submitting the review...

Deleting sections in a `uci:foreach` loop is not allowed and may lead to incorrect behaviour. Use `uci:delete_all` in its three-argument form (see `package/gluon-mesh-vpn-fastd/luasrc/lib/gluon/upgrade/400-mesh-vpn-fastd` for an example).

Reply:
I just looked at the libuci code, and deleting sections in a foreach loop should be safe after all. I vaguely remember a similar problem from a few years back, but it might have been a different issue. Switching to `uci:delete_all` would still make the code a bit shorter.

Reply:
It seems my first comment was right after all: I still get reproducible segfaults when I try to delete sections in a `uci:foreach` call.

Reply:
Interestingly this code worked for me. Still I adjusted it.